src/NServiceBus.Transport.Msmq/MsmqSettings.cs

```diff
@@ -50,6 +51,11 @@ public MsmqSettings(ReadOnlySettings settings)
                 TimeToReachQueue = timeToReachQueue;
             }
 
+            if (settings.TryGet<TimeSpan>("MessageEnumeratorTimeout", out var messageEnumeratorTimeout))
```

Comment:
One last thought. I think we should do something like NserviceBus.Transport.Msmq.Key to make it clear it belongs to msmq